templates/resources/includes/deploy-task-engine-run-script.j2

```diff
@@ -107,47 +119,58 @@ if [ ! -f "${APP_HOME}/conf/{{ boot_conf }}" ]; then
       -e "s#\${HOSTNAME}#${HOSTNAME}#g" \
       -e "s#\${HOSTNAME_SUFFIX}#${HOSTNAME_SUFFIX}#g" \
       -e "s#\${SSL}#${SSL}#g" \
+      -e "s#\${SERVER_PORT}#${SERVER_PORT}#g" \
       -e "s#\${APP_PORT}#${APP_PORT}#g" \
       ${APP_HOME}/default-conf/{{ boot_conf }}.template > ${APP_HOME}/conf/{{ boot_conf }}
 
     echo "Done"
   fi
 else
     echo "Found ${APP_HOME}/conf/{{ boot_conf }} file. Processing it for new properties"
+
+    if [[ -z "$OVERRIDE_HOSTNAME" ]]; then
+      HOSTNAME=$(hostname)
+      if [[ ${USE_IP_AS_HOSTNAME,,} == "true" ]]; then
+        HOSTNAME=$(hostname -i)
+      fi
+    else
+      HOSTNAME=$OVERRIDE_HOSTNAME
+    fi
+
     if [[ "${HOSTNAME}" != "" && "${HOSTNAME_SUFFIX}" != "" ]]; then
-      echo "Updating HOSTNAME & HOSTNAME_SUFFIX properties"
-      sed -i "s#server.hostname\=.*##" ${APP_HOME}/conf/{{ boot_conf }}
+      echo "Updating server.hostname with HOSTNAME & HOSTNAME_SUFFIX properties"
+      grep "server.hostname=" ${APP_HOME}/conf/{{ boot_conf }} && sed -i "s#server.hostname\=.*##" ${APP_HOME}/conf/{{ boot_conf }}
       {
         echo server.hostname=${HOSTNAME}${HOSTNAME_SUFFIX}
       } >> ${APP_HOME}/conf/{{ boot_conf }}
     fi
```

Comment:
```
 sed -i "s/server.hostname=.*/server.hostname=${HOSTNAME}${HOSTNAME_SUFFIX}/"
```
This should do the job. (Maybe we can refactor it later. Its not in the scope of this story and moreover there are many occurences)

Reply:
Check above comment

Reply:
The suggested sed command is a condensed version of commands from L142 to L145.